services/distribution/src/main/java/app/coronawarn/server/services/distribution/objectstore/client/ObjectStoreClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Corona-Warn-App
+ *
+ * SAP SE and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package app.coronawarn.server.services.distribution.objectstore.client;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Simple Storage Service (aka S3) client to perform bucket and object operations.
+ */
+public interface ObjectStoreClient {
+
+  /**
+   * Downloads the all objects that match the specified prefix from the specified object store bucket.
+   *
+   * @param bucket The name of the object store bucket.
+   * @param prefix The prefix that the names of the returned objects start with.
+   * @return A list of objects from the object store that match the specified parameters.
+   * @throws ObjectStoreOperationFailedException if the operation could not be performed.
+   */
+  List<S3Object> getObjects(String bucket, String prefix) throws ObjectStoreOperationFailedException;
```

Comment:
FYI @michael-burwig  Sonar doesn't like this - https://sonarci.wdf.sap.corp:8443/sonar/project/issues?id=app.coronawarn&open=AXJg1tQYhDC-tW6ttbKs&resolved=false